Apps/W1/HybridGP/app/src/codeunits/GPCloudMigration.codeunit.al

```diff
@@ -139,14 +132,42 @@ codeunit 4025 "GP Cloud Migration"
     local procedure CreateDataMigrationEntites(var DataMigrationEntity: Record "Data Migration Entity"): Boolean
     var
         HelperFunctions: Codeunit "Helper Functions";
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
     begin
         DataMigrationEntity.InsertRecord(Database::"G/L Account", HelperFunctions.GetNumberOfAccounts());
-        DataMigrationEntity.InsertRecord(Database::Customer, HelperFunctions.GetNumberOfCustomers());
-        DataMigrationEntity.InsertRecord(Database::Vendor, HelperFunctions.GetNumberOfVendors());
-        DataMigrationEntity.InsertRecord(Database::Item, HelperFunctions.GetNumberOfItems());
+
+        if GPCompanyAdditionalSettings.GetReceivablesModuleEnabled() then
+            DataMigrationEntity.InsertRecord(Database::Customer, HelperFunctions.GetNumberOfCustomers());
+
+        if GPCompanyAdditionalSettings.GetPayablesModuleEnabled() then
+            DataMigrationEntity.InsertRecord(Database::Vendor, HelperFunctions.GetNumberOfVendors());
+
+        if GPCompanyAdditionalSettings.GetInventoryModuleEnabled() then
+            DataMigrationEntity.InsertRecord(Database::Item, HelperFunctions.GetNumberOfItems());
+
         exit(true);
     end;
 
+    local procedure CreateConfiguredDataMigrationStatusRecords(var DataMigrationEntity: Record "Data Migration Entity")
```

Comment:
Are these record counts valid based upon selections made in the UI?

Reply:
Yes

Reply:
Actually, I do need to add in the updated count logic because disabling modules is part of this PR.